crates/edr_solidity/src/contracts_identifier/radix_tree.rs

```diff
@@ -0,0 +1,416 @@
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct RadixNode {
+    content: Vec<u8>,
+    is_present: bool,
+    bytes_matched_before: usize,
+    child_nodes: HashMap<u8, RadixNode>,
+}
+
+impl RadixNode {
+    fn new(content: Vec<u8>, is_present: bool, bytes_matched_before: usize) -> RadixNode {
+        RadixNode {
+            content,
+            is_present,
+            bytes_matched_before,
+            child_nodes: HashMap::new(),
+        }
+    }
+
+    pub fn get_content(&self) -> &Vec<u8> {
+        &self.content
+    }
+
+    fn add_word(&mut self, word: Vec<u8>) {
+        if word.is_empty() {
+            return;
+        }
+
+        let b = word[0];
+
+        // we temporarily remove the next node and then insert it back, possibly mutated and/or in a different position
+        let next_node = self.child_nodes.remove(&b);
+
+        match next_node {
+            None => {
+                let bytes_matched_before = self.bytes_matched_before + self.content.len();
+
+                let node = RadixNode::new(word, true, bytes_matched_before);
```

Comment:
There's no need to use a `match` for option, as you can do `if let Some(mut next_node) = next_node { .. } else { .. }`

Reply:
Done, thanks